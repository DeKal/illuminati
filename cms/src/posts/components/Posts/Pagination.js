import React from 'react'
import PropTypes from 'prop-types'
import TablePagination from '@material-ui/core/TablePagination'
import { calcMaxRowsPerPage } from 'core/utils/misc'

const Pagination = ({
  postCount,
  rowsPerPage,
  page,
  changePage,
  changeRowsPerPage
}) => (
  <TablePagination
    rowsPerPageOptions={calcMaxRowsPerPage(postCount)}
    component="div"
    count={postCount}
    rowsPerPage={rowsPerPage}
    page={page}
    onChangePage={(_, newPage) => changePage(newPage)}
    onChangeRowsPerPage={event =>
      changeRowsPerPage(parseInt(event.target.value, 10))
    }
  />
)

Pagination.propTypes = {
  postCount: PropTypes.number,
  rowsPerPage: PropTypes.number,
  page: PropTypes.number,
  changePage: PropTypes.func,
  changeRowsPerPage: PropTypes.func
}

export default Pagination
