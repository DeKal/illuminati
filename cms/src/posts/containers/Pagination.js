import { connect } from 'react-redux'
import Pagination from 'posts/components/Posts/Pagination'
import { changePage, changeRowsPerPage } from 'posts/state/actions'
import { getPage, getRowsPerPage } from 'posts/state/selectors'

const mapStateToProps = state => ({
  page: getPage(state),
  rowsPerPage: getRowsPerPage(state)
})

const mapDispatchToProps = {
  changePage,
  changeRowsPerPage
}

export default connect(mapStateToProps, mapDispatchToProps)(Pagination)
