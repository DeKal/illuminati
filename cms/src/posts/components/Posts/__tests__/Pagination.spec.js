import * as React from 'react'
import { shallow, mount } from 'enzyme'
import Pagination from '../Pagination'

describe('Pagination', () => {
  it('should render with no error', function() {
    const wrap = shallow(<Pagination postCount={20} rowsPerPage={5} page={0} />)
    expect(wrap).toMatchSnapshot()
  })

  it('should trigger changePage when press next page', function() {
    const changePage = jest.fn()
    const wrap = mount(
      <Pagination
        postCount={20}
        rowsPerPage={5}
        page={0}
        changePage={changePage}
      />
    )
    wrap
      .find("WithStyles(ForwardRef(IconButton))[title='Next page']")
      .simulate('click')
    expect(changePage.mock.calls.length).toBe(1)
  })
})
