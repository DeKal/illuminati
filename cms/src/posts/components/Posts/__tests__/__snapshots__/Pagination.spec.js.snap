// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render with no error 1`] = `
<WithStyles(ForwardRef(TablePagination))
  component="div"
  count={20}
  onChangePage={[Function]}
  onChangeRowsPerPage={[Function]}
  page={0}
  rowsPerPage={5}
  rowsPerPageOptions={
    Array [
      5,
      10,
      25,
    ]
  }
/>
`;
